components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/ContentPermissionDialogFragment.kt

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.sitepermissions
+
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import mozilla.components.browser.session.Session
+import mozilla.components.concept.engine.permission.Permission
+
+/**
+ * This is a general representation of a dialog meant to be used in collaboration with
+ * [SitePermissionsFeature] to show a dialog when a [Session.contentPermissionRequest]
+ *  is requested. If [SimplePermissionDialogFragment] is not flexible enough for your use case
+ *  you should inherit from this class and create your custom dialog.
+ */
+abstract class ContentPermissionDialogFragment : DialogFragment() {
```

Comment:
I'm confused about why you went ahead and implemented dialogs here? Didn't we plan to use doorhangers?!

Reply:
Sorrry, my bad, I thought that we only going to use  doorhangers on the toolbar, I will switch the implentation to use doorhangers!

Reply:
Maybe we can land this feature component without the content permission part; so that it only contains the foundation and the app permission part? Is that easy to detach?

I'm trying to get `DoorhangerPrompt` done today. Currently writing tests. :)

Reply:
Sounds good to me! 👍